spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/s3/S3AutoConfiguration.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.awspring.cloud.autoconfigure.s3;
+
+import java.util.Optional;
+
+import io.awspring.cloud.core.SpringCloudClientConfiguration;
+import io.awspring.cloud.s3.CrossRegionS3Client;
+import io.awspring.cloud.s3.S3ProtocolResolver;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.regions.providers.AwsRegionProvider;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.S3ClientBuilder;
+import software.amazon.awssdk.services.s3.S3Configuration;
+
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.util.StringUtils;
+
+/**
+ * {@link EnableAutoConfiguration} for {@link S3Client} and {@link S3ProtocolResolver}.
+ *
+ * @author Maciej Walkowiak
+ */
+@ConditionalOnClass({ S3Client.class, CrossRegionS3Client.class })
+@EnableConfigurationProperties(S3Properties.class)
+@Configuration(proxyBeanMethods = false)
+@Import(S3ProtocolResolver.class)
+public class S3AutoConfiguration {
```

Comment:
should we have `spring.cloud.aws.s3.enabled` ConditionOnProperty?

Reply:
👍